### PR TITLE
Feature/81 producer page

### LINKED ISSUE #81: Implement producer page

### PR DESCRIPTION
**Pull request by Moltenship**

### CODE REVIEW

File: src/components/EntityTabs/EntityTabs.tsx
Comment (moetazaneta) [inline src/components/EntityTabs/EntityTabs.tsx]: Missing readonly

File: src/components/EntityTabs/EntityTabs.tsx
Comment (moetazaneta) [inline src/components/EntityTabs/EntityTabs.tsx]: Perhaps, you should use `useCallback`

File: src/components/EntityTabs/EntityTabs.tsx
Comment (moetazaneta) [inline src/components/EntityTabs/EntityTabs.tsx]: Remove paddings, main layout will be responsible for horizontal paddings

File: src/components/Releases/Releases.tsx
Comment (moetazaneta) [inline src/components/Releases/Releases.tsx]: `readonly [language: string]: readonly string[];`

File: src/components/Releases/Releases.tsx
Comment (moetazaneta) [inline src/components/Releases/Releases.tsx]: Add readonly for `Release[]`

File: src/components/Releases/Releases.tsx
Comment (moetazaneta) [inline src/components/Releases/Releases.tsx]: Great use of generic here! 😽

File: src/features/Producer/components/ProducerRelations/ProducerRelations.tsx
Comment (moetazaneta) [inline src/features/Producer/components/ProducerRelations/ProducerRelations.tsx]: Is it possible to add Readonly for Record?

File: src/features/Producer/components/ProducerRelations/ProducerRelations.tsx
Comment (moetazaneta) [inline src/features/Producer/components/ProducerRelations/ProducerRelations.tsx]: What is `prod`? I would suggest `p` or `producer`

File: src/models/links.ts
Comment (moetazaneta) [inline src/models/links.ts]: Empty returns

**Review by moetazaneta — APPROVED** [review]

File: src/components/EntityTabs/EntityTabs.tsx
Comment (NightOfNights) [inline src/components/EntityTabs/EntityTabs.tsx]: Let's use `handle` prefix here

File: src/components/EntityTabs/EntityTabs.tsx
Comment (NightOfNights) [inline src/components/EntityTabs/EntityTabs.tsx]: Move this function outside of component scope, since it doesn't depend on any state

File: src/components/EntityTabs/EntityTabs.tsx
Comment (NightOfNights) [inline src/components/EntityTabs/EntityTabs.tsx]: `tabIndx` -> `tabIndex`

File: src/components/Releases/Releases.tsx
Comment (NightOfNights) [reply src/components/Releases/Releases.tsx]: It's the old code 🤡

Comment (NightOfNights) [reply src/components/Releases/Releases.tsx]: Btw, I think this function should be moved out of component scope too

File: src/components/Releases/Releases.tsx
Comment (NightOfNights) [inline src/components/Releases/Releases.tsx]: Same

File: src/features/Producer/components/ProducerRelations/ProducerRelations.tsx
Comment (NightOfNights) [inline src/features/Producer/components/ProducerRelations/ProducerRelations.tsx]: The more I see things like this in our components that combines/groupes some domain entities together, the more I think we are doing something wrong 😖. Components shouldn't do that, they should just display some UI things and work with their inner state. Well, I think our domain layer is just a big mess, that needs some thoughtful refactor, like what I did with links in https://github.com/vndbre/vndbre/pull/135

Well, at least, let's move this out of component scope as a function that has `producer` and `relatedProducers` params

File: src/features/Producer/components/ProducerRelations/ProducerRelations.tsx
Comment (NightOfNights) [inline src/features/Producer/components/ProducerRelations/ProducerRelations.tsx]: Extra fragment

File: src/features/VisualNovel/utils/visualNovelRoutesInfo.ts
Comment (NightOfNights) [inline src/features/VisualNovel/utils/visualNovelRoutesInfo.ts]: Is there any reason to move out this to `utils`? Let's keep it in `VisualNovelPage.tsx`

File: src/models/links.ts
Comment (NightOfNights) [inline src/models/links.ts]: Please, wait until #135 is merged. I refactored links stuff

File: src/features/Producer/utils/producerRouteParams.ts
Comment (NightOfNights) [inline src/features/Producer/utils/producerRouteParams.ts:1]: Let's keep it inside `Producer/routes.tsx`, so in case if params are changed, the developer would notice and update this interface with a higher chance :)

**Review by NightOfNights — CHANGES_REQUESTED** [review]

File: src/components/EntityTabs/EntityTabs.tsx
Comment (NightOfNights) [inline src/components/EntityTabs/EntityTabs.tsx]: Btw, don't forget to delete `VisualNovelTabs`

File: src/components/Releases/Releases.tsx
Comment (NightOfNights) [reply src/components/Releases/Releases.tsx]: `releasesData` -> `data` or `items`

File: src/components/Releases/Releases.tsx
Comment (NightOfNights) [reply src/components/Releases/Releases.tsx]: `ReactElement` -> `ReactNode`

File: src/features/Producer/utils/producerRouteParams.ts
Comment (Moltenship) [reply src/features/Producer/utils/producerRouteParams.ts:1]: It's related to dependency cycle import

File: src/features/Producer/components/ProducerRelations/ProducerRelations.tsx
Comment (Moltenship) [reply src/features/Producer/components/ProducerRelations/ProducerRelations.tsx]: It is intended

File: src/api/services/releasesService.ts
Comment (NightOfNights) [inline src/api/services/releasesService.ts]: Please, use `function declaration`

File: src/api/services/producersService.ts
Comment (NightOfNights) [inline src/api/services/producersService.ts]: Please, use `function declaration`

File: src/api/mappers/producerMapper.ts
Comment (NightOfNights) [inline src/api/mappers/producerMapper.ts]: You should have chosen incoming changes over your current on merge with `develop` 😭 I slightly improved docs, replaced with `arrow functions` with `function declaration` and moved out of `namespace` non-export things.

**Review by NightOfNights — APPROVED** [review]

File: src/api/mappers/producerMapper.ts
Comment (Moltenship) [reply src/api/mappers/producerMapper.ts]: :( I'll fix it

**Pull request merged**
